### PR TITLE
Feature/login popup

### PR DESCRIPTION
**Pull request by eloigrndl**

### CODE REVIEW

File: app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java [hunk 1]
```diff
@@ -100,7 +115,7 @@ public void signInFragmentUnsuccessfulLoginTest(){
 
         verify(mockedAuth).logIn(anyString(), anyString());
 
-        onView(withId(R.id.exceptionSignIn)).check(matches(withText(containsString(unsuccessfulLogin))));
+        onView(withId(android.R.id.button1)).check(matches(isClickable()));
```
Comment (khcr) [inline app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java:118]: The id button1 is not meaningful.

File: app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java [hunk 2]
```diff
@@ -201,11 +215,11 @@ public void signUpFragmentUnsuccessfulRegisterTest(){
...
 
-        onView(withId(R.id.exceptionSignUp)).check(matches(withText(containsString(unsuccessfulRegister))));
+        onView(withId(android.R.id.button1)).check(matches(isClickable()));
     }
 
     @Test
-    public void signUpFragmentUnsuccessfulRegisterTest2(){
+    public void signUpFragmentSuccessfulRegisterTest(){
         CompletableFuture<AuthenticationResponse> future = new CompletableFuture<>();
         future.complete(new AuthenticationResponse(true, null, null));
```
Comment (khcr) [inline app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java:224]: Could use CompletableFuture.completedFuture(...) for convenience

File: app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java [hunk 3]
```diff
@@ -219,7 +233,26 @@ public void signUpFragmentUnsuccessfulRegisterTest2(){
...
+                description.appendText("has no error text: ");
+            }
+
+            @Override
+            protected boolean matchesSafely(EditText view) {
+                return view.getError() == null;
+            }
+        };
+    }
+
```
Comment (khcr) [inline app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java:257]: Wouldn't it more robust to match against an error text ?

**Review by khcr — APPROVED** [review]
Nice and sweet. I approve the new pop-ups and the refactoring done.

File: app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java (hunk 1 above)
Comment (eloigrndl) [reply app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java:118]: It's the default id for the button of the pop-up I do not think I can change it

Comment (khcr) [reply app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java:118]: Fine then, my bad

File: app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java (hunk 2 above)
Comment (eloigrndl) [reply app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java:224]: Yes I forgot this when I was coding I guess

File: app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java (hunk 3 above)
Comment (eloigrndl) [reply app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java:257]: I only use it to check that the error field of some component is empty so I don't think we need to

File: app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java (hunk 2 above)
Comment (eloigrndl) [reply app/src/androidTest/java/com/ncnf/authentication/LoginActivityTest.java]: fixed

**Review by emmadeno — APPROVED** [review]
Looks good !

**Pull request merged**